core/src/main/java/org/bitcoinj/store/SPVBlockStore.java

```diff
@@ -111,9 +113,10 @@ public SPVBlockStore(NetworkParameters params, File file) throws BlockStoreExcep
      * @param grow wether or not to migrate an existing block store of different capacity
      * @throws BlockStoreException if something goes wrong
      */
-    public SPVBlockStore(NetworkParameters params, File file, int capacity, boolean grow) throws BlockStoreException {
+    public SPVBlockStore(Block genesisBlock, File file, int capacity, boolean grow) throws BlockStoreException {
         Objects.requireNonNull(file);
-        this.params = Objects.requireNonNull(params);
+        this.genesisHeader = Objects.requireNonNull(genesisBlock).cloneAsHeader();
+        this.serializer = genesisHeader.getParams().getDefaultSerializer(); // TODO: Better way of getting serializer?
```

Comment:
I think this line makes this PR a net loss. We want to get rid of both params and serializer in `Block` (genesisBlock).